### ISSUE TITLE
Release changelog

### ISSUE THREAD

**Issue by goeddea** [crossbario/crossbar#756]
As part of a proper, more dependable release process, we should provide a release changelog.

migrated from https://github.com/crossbario/crossbardocs/issues/55

**Comment by agronholm** [comment]
When I fix a bug or add a new feature to my own projects, I usually edit the changelog in the same commit. Perhaps you should do the same?

**Comment by meejah** [comment]
We've been trying to use "town crier": the way this is supposed to work is that you put a new file in `crossbar/newsfragments` and then right before a new release, you run `towncrier` (actually `make news`) which should incorporate all the fragments into the latest `docs/pages/ChangeLog.md` and make a git commit deleting all the (now used-up) newsfragments -- **however** this needs to be better-incorporated into our releases.

I think really what this bug needs out of it is "better release automation". Another nice-to-have would be some sort of GitHub hook that checks each PR and warns (or errors?) if there's no new newsfragment for it.

**Comment by agronholm** [comment]
I see the directory only has 4 news items from a time period of 8 months. Sorry to say but doesn't look like people have been trying much.

**Comment by oberstet** [comment]
@agronholm I am guilty here, most definitely. By not using it=( Probably we need to enforce it via CI like @meejah hints at ..

**Comment by meejah** [comment]
So I can think of two pretty simple things we could do here to make this "easier to do":

- [ ] a git alias / script to create a bug-branch given only a ticket number (which can then also add a default/placeholder newsfragment for you)
- [ ] CI check that doesn't pass unless there's a `<ticketnumber>.something` in the newsfragment directory of your PR (would be easy with a git hook, but not sure how hard this is to do within github, maybe someone's already built something?)

**Comment by agronholm** [comment]
I just evaluated towncrier today and concluded that its model was too rigid for me. Every single change would require a ticket to be opened. I'm not convinced that my model of amending the changelog within the relevant commit is any worse than putting each changelog entry in its separate file. It's certainly simpler.

**Comment by meejah** [comment]
`towncrier` came from Twisted, which does require a ticket for every change.

In any case, a "CI hook" would be great -- whether it checks for the right ``newsfragment`` or checks that you edited ``ChangeLog`` or similar is really just a detail...

**Comment by rrueth** [comment]
Has there been any progress on maintaining a changelog?  For example, v18.11.1 was released today, and I don't know how to find what has changed in it without going through all of the commits.  We've been pretty hesitant to update our Crossbar deployment without knowing what has changed.

**Comment by gregkeys** [comment]
Agreed, we need a changelog, Im getting dropped endpoints and I'm not sure if I need to update or debug. a changelog would help

**Comment by rrueth** [comment]
Can you please add a changelog for the v19.2.1 release?

**Comment by rrueth** [comment]
@oberstet Can you please add a changelog for v19.6.1?  It makes it very difficult for us to know what to expect when upgrading without a changelog.
